### PR TITLE
feat: further bbb options from #96

### PR DESCRIPTION
**Pull request by mschwrdtnr**
I implemented the bbb options from #96 into the current version 2.3. For cron I didn't found a template file or something like.

Let me know if you have further improvements for this.

EDIT: This also implements some new options for breakout rooms and a template for `cron.daily` to set retention period.

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
-bbb_hostname: "{{ ansible_fqdn }}"
```
Comment (BrutalBirdie) [inline defaults/main.yml:2]: Why did you change all double quotes to single quotes?
Just curious.

File: tasks/config.yml [hunk 2]
```diff
@@ -293,3 +293,27 @@
     dest: "{{ bbb_nginx_root | regex_replace('\\/$', '') }}/{{ bbb_custom_presentation_name | default(bbb_custom_presentation) }}"
     mode: '0644'
   when: bbb_custom_presentation is defined
+
+  - name: Set the retention period for presentations, red5, kurento, and freeswitch caches
```
Comment (BrutalBirdie) [inline tasks/config.yml:297]: red5 does no longer exist as far as I am aware of?

File: tasks/config.yml [hunk 3]
```diff
@@ -293,3 +293,27 @@
...
+  lineinfile:
```
Comment (BrutalBirdie) [inline tasks/config.yml:298]: Instead of working with so many `lineinfile` replacement and RegEx I would prefer starting to template this file.

File: templates/conference.conf.xml [hunk 4]
```diff
@@ -69,9 +69,18 @@
...
+      {% else %}
+      <!-- <param name="muted-sound" value="conference/conf-muted.wav"/> -->
+      {% endif %}
+
       <!-- File to play to acknowledge unmuted -->
+      {% if bbb_freeswitch_unmuted_sound is true %}
       <param name="unmuted-sound" value="conference/conf-unmuted.wav"/>
+      {% else %}
+      <!-- <param name="unmuted-sound" value="conference/conf-unmuted.wav"/> -->
+      {% endif %}
```
Comment (BrutalBirdie) [inline templates/conference.conf.xml:83]: I think this will lead to bad/wrong indentation.
See https://svn.python.org/projects/external/Jinja-1.1/docs/build/escaping.html
Can also be seen in #112

**Review by BrutalBirdie — CHANGES_REQUESTED** [review]

File: defaults/main.yml (hunk 1 above)
Comment (mschwrdtnr) [reply defaults/main.yml:2]: Honestly at first my prettier plugin changed to single quotes. but after I rechecked the file I saw that this file partially uses single quotes and partially double quotes e.g: https://github.com/ebbba-org/ansible-role-bigbluebutton/blob/bbb/2.3/defaults/main.yml#L90-L94 and https://github.com/ebbba-org/ansible-role-bigbluebutton/blob/bbb/2.3/defaults/main.yml#L124-L126. Because of that I thought it would make more sense to use single quotes instead. I can change it back if you want.

**Comment by mschwrdtnr** [comment]
@BrutalBirdie thanks for you suggestions. I will implement this tomorrow.

**Comment by ChrisKar96** [comment]
Could I suggest adding the options for breakout rooms as well? I don't have a 2.3 installation yet to test, but in my 2.2 installations I needed

```
# Default settings for breakout rooms
breakoutRoomsEnabled={true|false}
breakoutRoomsRecord={true|false}
breakoutRoomsPrivateChatEnabled={true|false}
```

in /usr/share/bbb-web/WEB-INF/classes/bigbluebutton.properties

If not fit for current PR i will create one myself, as soon as I have a 2.3 installation so I can perform tests.

Thanks

File: defaults/main.yml (hunk 1 above)
Comment (BrutalBirdie) [reply defaults/main.yml:2]: I guess you could say it's flavor double quotes are need when you do something like this:
```
"'service_name' in groups"
```

**Comment by BrutalBirdie** [comment]
> Could I suggest adding the options for breakout rooms as well? I don't have a 2.3 installation yet to test, but in my 2.2 installations I needed
> 
> ```
> # Default settings for breakout rooms
> breakoutRoomsEnabled={true|false}
> breakoutRoomsRecord={true|false}
> breakoutRoomsPrivateChatEnabled={true|false}
> ```
> 
> in /usr/share/bbb-web/WEB-INF/classes/bigbluebutton.properties
> 
> If not fit for current PR i will create one myself, as soon as I have a 2.3 installation so I can perform tests.
> 
> Thanks

Yes this could fit but keep in mind we override the `/usr/share/bbb-web/WEB-INF/classes/bigbluebutton.properties` file with `/etc/bigbluebutton/bbb-web.properties`.

File: tasks/config.yml (hunk 3 above)
Comment (mschwrdtnr) [reply tasks/config.yml:298]: I don't really know what the best practice for templating this file would be. 
Is it just to take [the file from bbb](https://github.com/bigbluebutton/bigbluebutton/blob/develop/bigbluebutton-config/cron.daily/bigbluebutton) and template the vars we want to set dynamically? If so, I would also create a new directory `bbb-config/cron.daily`.

Comment (BrutalBirdie) [reply tasks/config.yml:298]: `templates/bbb-config/cron.daily` would be OK for me.
Please add the URL to the file into the template as a comment.
```
# https://github.com/ebbba-org/ansible-role-bigbluebutton/blob/bbb/2.3/tasks/kurento.yml
```
```
history=5
unrecorded_days=14
published_days=14
log_history=28
```
For now I think only this part needs to be templated.
Something like this - open for others vars
```
history={{ bbb_cron_history }}
unrecorded_days={{ bbb_cron_unrecorded_days }}
published_days={{ bbb_cron_published_days }}
log_history={{ bbb_cron_log_history }}
```
And using the defaults from the file as default in the vars.

**Comment by mschwrdtnr** [comment]
@BrutalBirdie now everything should be fine.

I also added your suggestions @ChrisKar96.

File: defaults/main.yml [hunk 5]
```diff
@@ -108,6 +110,22 @@ bbb_ntp_cron_day: "*"
...
+bbb_default_meeting_duration: 0
+bbb_max_num_pages: 200
+bbb_max_conversion_time: 5
+bbb_num_conversion_threads: 5
+bbb_num_file_processor_threads: 2
+
+bbb_breakout_rooms_enabled: true
+bbb_breakout_rooms_record: false
+bbb_breakout_rooms_privatechat_enabled: true
+
```
Comment (BrutalBirdie) [inline defaults/main.yml:128]: Isn't a big part of this missing in the README?

File: tasks/config.yml [hunk 6]
```diff
@@ -293,3 +293,15 @@
...
+  become: true
+  template:
+    src: bbb-config/cron.daily/bigbluebutton
+    dest: /etc/cron.daily/bigbluebutton
+    mode: '0644'
+    owner: root
+    group: root
+  notify:
+    - reload systemd
+    - restart bigbluebutton
```
Comment (BrutalBirdie) [inline tasks/config.yml:307]: yamllint - no new line in end of file.

File: templates/bbb-config/cron.daily/bigbluebutton [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+# version.
+#
+# BigBlueButton is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with BigBlueButton; if not, see <http://www.gnu.org/licenses/>.
+# 
+# File: https://github.com/bigbluebutton/bigbluebutton/blob/develop/bigbluebutton-config/cron.daily/bigbluebutton
```
Comment (BrutalBirdie) [inline templates/bbb-config/cron.daily/bigbluebutton:20]: 👍🏼

File: templates/conference.conf.xml [hunk 8]
```diff
@@ -69,9 +69,17 @@
...
       <param name="muted-sound" value="conference/conf-muted.wav"/>
+      {% else -%}
+      <!-- <param name="muted-sound" value="conference/conf-muted.wav"/> -->
+      {% endif -%}
       <!-- File to play to acknowledge unmuted -->
+      {% if bbb_freeswitch_unmuted_sound == true -%}
       <param name="unmuted-sound" value="conference/conf-unmuted.wav"/>
+      {% else -%}
+      <!-- <param name="unmuted-sound" value="conference/conf-unmuted.wav"/> -->
+      {% endif -%}
```
Comment (BrutalBirdie) [inline templates/conference.conf.xml:82]: Are you sure this is the right indentation?
See:
https://svn.python.org/projects/external/Jinja-1.1/docs/build/escaping.html
#112

**Review by BrutalBirdie — CHANGES_REQUESTED** [review]

File: templates/conference.conf.xml (hunk 8 above)
Comment (mschwrdtnr) [reply templates/conference.conf.xml:82]: Yes I am sure. I tested it several times by setting the vars true/false.

At first I didn't used the `-`, see https://github.com/ebbba-org/ansible-role-bigbluebutton/pull/115/commits/626b5de7489df86599b062cea86d50f0fd8ee15c but I think it looks better with the minus.

File: README.md [hunk 9]
```diff
@@ -83,18 +83,33 @@ Also check [Before you install](https://docs.bigbluebutton.org/2.2/install.html#
...
 | `bbb_container_compat` | Compatibility with unprivileged containers | `false` | Enabling this option allows to deploy BBB into a unprivileged container |
 | `bbb_firewall_ufw` | A dict of rules for the ufw | see `defaults/main.yml` | can also be used to allow/deny more/less |
+| `bbb_default_max_users` | Default maximum number of users a meeting can have | `0` | Meeting doesn't have a user limit |
+| `bbb_default_meeting_duration` | Default duration of the meeting in minutes | `0` | Meeting doesn't end |
+| `bbb_max_num_pages` | Maximum number of pages allowed for an uploaded presentation | `200` | |
+| `bbb_max_conversion_time` | Number of minutes the conversion should take | `5` | If it takes more than this time, cancel the conversion process |
+| `bbb_num_conversion_threads` | Number of threads in the pool to do the presentation conversion | `5` | |
+| `bbb_num_file_processor_threads` | Number of threads to process file uploads | `2` | |
+| `bbb_freeswitch_muted_sound` | Enable muted sound (`you are now muted`) | `true` | |
+| `bbb_freeswitch_unmuted_sound` | Enable unmuted sound (`you are now unmuted`) | `true` | |
```
Comment (BrutalBirdie) [inline README.md:109]: This is for the SIP mute/unmute sound right?

**Review by BrutalBirdie — APPROVED** [review]
Thanks for the great work!
I've found nothing else to nag about. >)

Thanks for the work and patience.

`#APPROVED`

File: README.md (hunk 9 above)
Comment (mschwrdtnr) [reply README.md:109]: If I am right, this vars are for the standard mute/unmute button sounds. I am not sure if SIP is envolved there.

Comment (BrutalBirdie) [reply README.md:109]: Ahhh yes you are right.

**Comment by Davka** [comment]
Some changes overlap with # 110 other missing items and will be submitted with # 110. Would be great if that were merged.

**Comment by anadahz** [comment]
@mschwrdtnr Thank you for working on this PR.
The PR looks good to be merged. One more reviewer with write access is required to merge the PR.

Unrelated to this PR how many people has write access to this repository?

**Comment by BrutalBirdie** [comment]
> @mschwrdtnr Thank you for working on this PR.
> The PR looks good to be merged. One more reviewer with write access is required to merge the PR.
> 
> Unrelated to this PR how many people has write access to this repository?

three is the current stand.

**Comment by mschwrdtnr** [comment]
@BrutalBirdie what's the problem with this PR? Should I merge the changes with @Davka?

I would say let's merge #110 and I will pull the new state and make the changes that are necessary or overlapping.

**Review by Davka — APPROVED** [review]

**Review by Davka — APPROVED** [review]

**Review by Davka — APPROVED** [review]
